bolt/lib/Core/ParallelUtilities.cpp

```diff
@@ -102,12 +102,15 @@ inline unsigned estimateTotalCost(const BinaryContext &BC,
 
 } // namespace
 
-ThreadPoolInterface &getThreadPool() {
+ThreadPoolInterface &getThreadPool(const int ThreadsCount) {
```

Comment:
Make unsigned? Negative ThreadCount doesn't make sense.

Reply:
Wouldn't this be covered by the if statement? If it's less than 1 it just creates a SingleThreadExecutor. Unsigned would still allow for a ThreadCount that's 0 which also wouldn't work.

Reply:
Technically speaking yes, but just from API perspective < 0 number of threads is just not a valid value.